integrations/apollo/src/index.ts

```diff
@@ -56,19 +56,38 @@ export const createApolloClient = ({
   const uri = backendUrl
   const interpreter = nhost?.auth.client.interpreter
 
-  let token: string | null = null
+  let accessToken: AuthContext['accessToken'] | null = null
 
-  function getAuthHeaders() {
+  const isTokenValid = () =>
+    accessToken?.value && accessToken?.expiresAt && accessToken.expiresAt > new Date()
```

Comment:
```suggestion
    !!accessToken?.value && !!accessToken?.expiresAt && accessToken?.expiresAt > new Date()
```

There are cases (e.g: when `accessToken?.value` is falsy) where this function would not return a `boolean`.